File: js/test/NgramTest.js
const assert = require('chai').assert;
const Ngram = require('../Ngram.js').Ngram;
const Event = require('../Event.js').Event;

describe('Ngram', function() {
  describe('#constructor()', function() {
    it('should throw an exception with undefined eventList', () => {
        try  {
            let ng = new Ngram(undefined);
            assert.fail();
        } catch (ex) {

        }
    });
    it('should throw an exception with null item', () => {
        try  {
            let ng = new Ngram(null);
            assert.fail();
        } catch (ex) {

        }
    });
    it('should throw an exception with not an Array', () => {
        try  {
            let ng = new Ngram("hey");
            assert.fail();
        } catch (ex) {

        }
    });
    it('should throw an exception with not an Array of Event', () => {
        try  {
            let ng = new Ngram([1,2,3]);
            assert.fail();
        } catch (ex) {

        }
    });
    it('should hash empty', () => {
        let ng = new Ngram([]);
        assert.equal(ng.key,0);
    });
    it('should hash Event', () => {
        let e = new Event('this is my event');
        let ng = new Ngram([e]);
        assert.equal(ng.key,e.key);
    });
    it('should hash Event', () => {
        let e = new Event('this is my event');
        let ng = new Ngram([e,e]);
        assert.equal(ng.key,e.key+e.key);
    });
  });
});

File: java/src/test/java/com/naturalness/NGramTest.java
package com.naturalness;

import static org.junit.Assert.fail;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.assertEquals;


import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

public class NGramTest {

    @Test
    public void shouldCreateNGram() {
        try {
            new NGram(null);
            fail();
        } catch (Exception ex) {

        }   
    }

    @Test
    public void shouldCreateSameHash() {
        List<Event> eventList = new ArrayList<Event>();
        NGram ng1 = new NGram(eventList);
        NGram ng2 = new NGram(eventList);
        assertTrue(ng1.hashCode() == ng2.hashCode());
    }

    @Test
    public void shouldEncapsulateEventList() {
        List<Event> eventList = new ArrayList<Event>();
        NGram ng1 = new NGram(eventList);
        int hash = ng1.hashCode();
        eventList.add(new Event("hey"));
        assertTrue(ng1.hashCode() == hash);
    }

    @Test
    public void shouldCreateSameNGram() {
        List<Event> eventList = new ArrayList<Event>();
        NGram ng1 = new NGram(eventList);
        NGram ng2 = new NGram(eventList);
        assertEquals(ng1, ng2);
    }
}